Source/WebInspectorUI/UserInterface/Views/ColorSwatchDisplay.js

```diff
@@ -0,0 +1,59 @@
+WI.ColorSwatchDisplay = class ColorSwatchDisplay extends WI.Object
+{
+    constructor(delegate, colorPalette)
+    {
+        super();
+
+        this._delegate = delegate;
+
+        this._element = document.createElement("div");
+        this._element.classList.add("swatch-display");
+
+        let swatchesTitle = this._element.appendChild(document.createElement("h2"));
+        swatchesTitle.innerText = WI.UIString("Variables", "Variables", "Title of swatches section in Color Picker");
+
+        let swatchWrapper = this._element.appendChild(document.createElement("ul"));
+        this._sortedPalette = this._specializedSort(colorPalette);
+
+        for (let paletteObject of this._sortedPalette) {
+            let swatch = new WI.InlineSwatch(WI.InlineSwatch.Type.Color, paletteObject.color, {readOnly: true, swatchTooltip: paletteObject.colorSource});
+            swatch.element.addEventListener("click", (event) => {
+                this._delegate.colorSwatchDispalyColorSelected(this, paletteObject.color, paletteObject.colorSource);
```

Comment:
if we're requiring that `_delegate` be provided then please add this to the top of the `constructor`
```
console.assert(delegate && delegate.colorSwatchDisplayColorSelected, delegate);
```